### PR TITLE
cli: search @- for branches to push when @ has none

### PR DESCRIPTION
**Pull request by Ralith**
I went back and forth on whether scanning `@-` should be conditional on `@` being empty. Currently leaning towards "no" since you might have made some new changes before remembering you wanted to push, and since a nonempty new commit wouldn't land on a branch anyway unless the branch was explicitly updated.

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -4239,17 +4240,44 @@ fn cmd_git_push(
...
+                    view: &'a View,
+                    target: &RefTarget,
+                ) -> Vec<(&'a String, &'a BranchTarget)> {
+                    view.branches()
+                        .iter()
+                        .filter(|(_, branch_target)| {
+                            branch_target
+                                .local_target
+                                .as_ref()
+                                .map_or(false, |x| x == target)
```
Comment (martinvonz) [inline src/commands.rs:4253]: `branch_target.local_target.as_ref() == Some(target)` is a little simpler. I don't know which is faster, but speed probably doesn't matter here anyway. What do you think?

**Review by martinvonz — APPROVED** [review]
Thanks for this feature!

File: src/commands.rs (hunk 1 above)
Comment (Ralith) [reply src/commands.rs]: Yep, that's definitely better.

File: src/commands.rs [hunk 2]
```diff
@@ -4239,17 +4240,44 @@ fn cmd_git_push(
...
+
+                // Search for branches targeting @
+                let mut branches = find_branches_targeting(
+                    workspace_command.repo().view(),
+                    &RefTarget::Normal(checkout.clone()),
+                );
+                if branches.is_empty() {
+                    // Try @- instead if it has exactly one parent, such as after `jj squash`
+                    let commit = workspace_command.repo().store().get_commit(checkout)?;
+                    if let [parent] = commit.parent_ids() {
```
[the remark this answers is not in the document]
Comment (Ralith) [reply src/commands.rs:4263]: Slice patterns :tada:

**Pull request merged**